File: code/mathematical-algorithms/tribonacci_numbers/tribonacci.rs
// Prints series of 'n' tribonacci numbers 

fn tribonacci(n: i32) {
    let mut a = 0;
    let mut b = 1;
    let mut c = 1;
    print!("{}, {}, {}, ", a, b, c);
    for _ in 3..n {
        let d = a + b + c;
        a = b;
        b = c;
        c = d;
        print!("{}, ", d);
    }
}

fn main() {
    tribonacci(15)
}

File: cosmos-master/code/mathematical-algorithms/tribonacci_numbers/tribonacci.go
package main

import "fmt"

// Prints series of 'n' tribonacci numbers

func tribonacci(n int) {
	a, b, c := 0, 1, 1
	fmt.Printf("%d, %d, %d, ", a, b, c)
	for i := 3; i < n; i++ {
		d := a + b + c
		a, b, c = b, c, d
		fmt.Printf("%d, ", d)
	}
}

func main() {
	tribonacci(15)
}
